be/src/util/timezone_hsscan.h

```diff
@@ -0,0 +1,79 @@
+// This file is made available under Elastic License 2.0.
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/be/src/util/timezone_utils.h
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+#pragma once
+
+#include <hs/hs.h>
+#include <string.h>
+
+#include "common/status.h"
+
+// find time zone with hyper scan.
+namespace starrocks {
+class TimezoneHsScan {
+public:
+    // match example: +08:00、-08:00 etc.
+    const std::string hs_reg_pattern = "[+|-]{1}\\d{2}\\:\\d{2}";
```

Comment:
Is this regular expressoin maybe better start with '^' ? like it with RE2

Reply:
I changed same with re2.